### PR TITLE
 [fix][seekhandler] seek size can exceed min and max time of player

### PR DESCRIPTION
**Pull request by xhaggi**
The seek size in `CSeekHandler` could exceed the min and max time of the current playing item. This PR address it and also does some refactoring and cleanups as discussed with @FernetMenta.

@da-anda as you requested

### CODE REVIEW

File: xbmc/dialogs/GUIDialogSeekBar.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ void CGUIDialogSeekBar::FrameMove()
     return;
   }
 
-  unsigned int percent((!CSeekHandler::GetInstance().InProgress() && g_infoManager.GetTotalPlayTime())
+  unsigned int percent((!g_application.m_pPlayer->GetSeekHandler().InProgress() && g_infoManager.GetTotalPlayTime())
```
Comment (fritsch) [inline xbmc/dialogs/GUIDialogSeekBar.cpp]: That notation is a bit "odd" or let's say I am not used to of this multiple bool Int mix. Why not: unsigned int percentage = InProgress() ? GetTotalPlaytime : 0;

File: xbmc/ApplicationPlayer.h [hunk 2]
```diff
@@ -97,85 +86,101 @@ class CApplicationPlayer
...
 
-  protected:
-    std::shared_ptr<IPlayer> GetInternal() const;
+  CSeekHandler& GetSeekHandler();
+protected:
+  std::shared_ptr<IPlayer> GetInternal() const;
+
+private:
+  std::shared_ptr<IPlayer> m_pPlayer;
+  std::unique_ptr<CSeekHandler> m_seekHandler;
```
Comment (FernetMenta) [inline xbmc/ApplicationPlayer.h]: why is this a pointer and not a plain memeber?

File: xbmc/Application.cpp [hunk 3]
```diff
@@ -2911,7 +2911,7 @@ void CApplication::Stop(int exitCode)
     CServiceBroker::GetServiceAddons().Stop();
 
     // unregister action listeners
-    UnregisterActionListener(&CSeekHandler::GetInstance());
+    UnregisterActionListener(&g_application.m_pPlayer->GetSeekHandler());
```
Comment (da-anda) [inline xbmc/Application.cpp]: is `g_application.` really needed here?

Comment (xhaggi) [reply xbmc/Application.cpp]: Good catch, search/replace ;)

File: xbmc/dialogs/GUIDialogSeekBar.cpp (hunk 1 above)
Comment (xhaggi) [reply xbmc/dialogs/GUIDialogSeekBar.cpp]: Logic is untouched by me, should I change it to what you suggested in a separate commit?

File: xbmc/ApplicationPlayer.h (hunk 2 above)
Comment (xhaggi) [reply xbmc/ApplicationPlayer.h]: No reason, will change it

File: xbmc/dialogs/GUIDialogSeekBar.cpp (hunk 1 above)
Comment (xhaggi) [reply xbmc/dialogs/GUIDialogSeekBar.cpp]: after a second look it's a ternary operator.

```c++
unsigned int percent((!g_application.m_pPlayer->GetSeekHandler().InProgress() && g_infoManager.GetTotalPlayTime())
  ? lrintf(g_application.GetPercentage())
  : (unsigned int)g_infoManager.GetSeekPercent());
```
same like

```c++
unsigned int percent = !g_application.m_pPlayer->GetSeekHandler().InProgress() && g_infoManager.GetTotalPlayTime()
  ? lrintf(g_application.GetPercentage())
  : static_cast<unsigned int>(g_infoManager.GetSeekPercent());
```

**Comment by xhaggi** [comment]
@FernetMenta @da-anda addressed in the fixup commit, please take another look.

File: xbmc/dialogs/GUIDialogSeekBar.cpp (hunk 1 above)
Comment (fritsch) [reply xbmc/dialogs/GUIDialogSeekBar.cpp]: Ah! Thanks much. Now it makes sense - no need to change anything, besides the global :-)

**Review by FernetMenta — APPROVED** [review]

File: xbmc/dialogs/GUIDialogSeekBar.cpp (hunk 1 above)
Comment (xhaggi) [reply xbmc/dialogs/GUIDialogSeekBar.cpp]: Np, not as easy to read I agree, because of the missing `=` and the surrounding braces ;)

**Pull request merged**